runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1202,6 +1202,8 @@ function M.set_loclist(opts)
 
   local open_loclist = if_nil(opts.open_loclist, true)
 
+  local win_id = vim.api.nvim_get_current_win()
+
   local bufnr = vim.api.nvim_get_current_buf()
```

Comment:
I think you can still have a race condition here, if somebody manages to switch windows between these two lines; assuming that's even possible.

Reply:
Between what two lines? I don't see any race condition here.

Reply:
@mjlbach Between these two:

```lua
local win_id = vim.api.nvim_get_current_win()

local bufnr = vim.api.nvim_get_current_buf()
```

I'm not sure if the user can actually do anything between running these two lines, but if they can (i.e. this runs in the background), the following could happen:

1. `local win_id = vim.api.nvim_get_current_win()` runs, produces window ID 1
2. The user switches to window 2
3. `local bufnr = vim.api.nvim_get_current_buf()` runs, producing the buffer ID for window 2; not window 1

The rest of the PR looks fine and works as far as I'm able to test.

Reply:
This isn't handled in a callback, this is user-invoked so it will be fine. Thanks for the review.